Geometry_Engine/Query/HashArray.cs

```diff
@@ -66,100 +86,173 @@ public static double[] IHashArray(this IGeometry igeometry)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(ICurve).IsAssignableFrom(t)) ?? false)
+                return default;
+
             translationFactor += (int)TypeTranslationFactor.Arc;
 
-            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor)
-               .Concat(curve.PointAtParameter(0.5).HashArray(translationFactor));
+            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor,
+                    comparisonConfig,
+                    fullName.AppendPropertyName($"{fullName}.{nameof(StartPoint)}"))
+               .Concat(curve.PointAtParameter(0.5).HashArray(translationFactor,
+                    comparisonConfig,
+                    fullName.AppendPropertyName($"{nameof(PointAtParameter)}(5e-1)"))
+               );
 
             if (!skipEndPoint)
-                hash = hash.Concat(curve.EndPoint().HashArray(translationFactor));
+                hash = hash.Concat(curve.EndPoint().HashArray(translationFactor, comparisonConfig, fullName.AppendPropertyName($"{nameof(EndPoint)}")));
 
             return hash.ToArray();
         }
 
         /***************************************************/
 
         [Description("The GeometryHash for an Circle is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Circle.")]
-        private static double[] HashArray(this Circle curve, double translationFactor)
+        private static double[] HashArray(this Circle curve, double translationFactor, BaseComparisonConfig comparisonConfig, string fullName = null)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(Circle).IsAssignableFrom(t)) ?? false)
+                return default;
+
             // The input `skipEndPoint` is not used here because Circles do not have a clearly defined endpoint to be used in a chain of segment curves.
 
             translationFactor += (int)TypeTranslationFactor.Circle;
 
-            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor)
-                   .Concat(curve.PointAtParameter(0.33).HashArray(translationFactor))
-                   .Concat(curve.PointAtParameter(0.66).HashArray(translationFactor));
+            IEnumerable<double> hash = curve.StartPoint().HashArray(translationFactor,
+                        comparisonConfig,
+                        fullName.AppendPropertyName($"{nameof(StartPoint)}"))
+                   .Concat(curve.PointAtParameter(0.33).HashArray(translationFactor,
+                        comparisonConfig,
+                        fullName.AppendPropertyName($"{nameof(PointAtParameter)}(33e-2)"))
+                   ).Concat(curve.PointAtParameter(0.66).HashArray(translationFactor,
+                        comparisonConfig,
+                        fullName.AppendPropertyName($"{nameof(PointAtParameter)}(66e-2)"))
```

Comment:
Following up ty my previous comment, here it is even more valid: the consumer would need to know the circles are sampled at 1/3, plus would need to know that a value needed to ignore it is 0.33 rather than 0.3333. Of course assuming that I have not misunderstood the code 😉

Reply:
Similar response as above, although here we took a more arbitrary decision. As per its description, the GeometryHash for a Circle is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Circle. This is because you need at least 3 points to make sure two Circles are equal. Also this is not the subject of this PR, which is again only adapting the existing mechanism to support BaseComparisonConfig. Happy for you to raise a separate issue if you think that needs improving.

Reply:
I do not question picking the points as unique definitions of curves. My question was pointed at the value of appending the name of such arbitrary property as point at 0.66 parameter - the consumer would never know that a point can be excluded from hash computation by setting such value in properties to ignore. Therefore, we would be fine with simply passing the original value of `fullName` I believe (assuming that no one cares about that name so deep down the algorithm). However, it probably makes sense to keep consistent across the entire call stack, so happy with the code as is! 👍

Reply:
Ah sorry I didn't understand! Yes you're right then -- it's may be a bit overzealous, I just went straight for the most flexible mechanism.